### ISSUE TITLE
Remove async features `stream` and `future` for Preview 2

### ISSUE THREAD

**Issue by Kylebrown9** [WebAssembly/wasi-filesystem#78]
If I remember correctly and it hasn't changed, wasi-snapshot-preview2 isn't going to rely on async features, but this repo still uses `stream` and `future` in its `.wit`.

**Comment by sunfishcode** [comment]
Yes. I'm expecting to do that once wit-bindgen supports `use` and we can start keeping the prototype world in separate files which can be more easily coordinated with the upstream proposal repos.

**Comment by Kylebrown9** [comment]
That makes sense, just wanted to make sure it was on your radar

**Comment by sunfishcode** [comment]
https://github.com/WebAssembly/wasi-proposal-template/pull/2 updates the wit-bindgen; if that's accepted, the next step is to start updating all the proposal repos.

**Comment by rektide** [comment]
not sure what impact this has on the ticket or what happens next with wasi-filesystem, but `stream` and `future` have landed in the component model.
https://github.com/WebAssembly/component-model/pull/405

**Comment by sunfishcode** [comment]
Indeed, and we're excited to adopt it. Some of the WASI proposals have added [wit-0.3.0-draft](https://github.com/WebAssembly/wasi-random/tree/main/wit-0.3.0-draft) directories showing what they'll look like with the new async features. I don't know if anyone's working on that for wasi-filesystem yet, but that would be one of the next steps.
